provider/azureProvider.js

```diff
@@ -178,7 +179,24 @@ return new BbPromise((resolve, reject) => {
       }
     }
 
-    const template = JSON.parse(fs.readFileSync(templateFilePath, 'utf8'));
+    let template = JSON.parse(fs.readFileSync(templateFilePath, 'utf8'));
+
```

Comment:
Instead of injecting environment variables into ARM template, can you add them variables list? See above for how userParameters are passed into ARM template?